src/algo_keys.c

```diff
@@ -3,40 +3,31 @@
 
 #include "algo_keys.h"
 
-// Allocate 64 bytes for privateKeyData because os_perso_derive_node_bip32
-// appears to write more than 32 bytes to this buffer.  However, we only
-// need 32 bytes for cx_ecfp_init_private_key..
-static uint8_t privateKeyData[64];
-
-uint8_t publicKey[32];
 
 void
-algorand_key_derive(void)
+algorand_key_derive(uint32_t accountId, cx_ecfp_private_key_t *privateKey)
 {
-  uint32_t bip32Path[5];
+  static uint8_t  privateKeyData[64];
+  static uint32_t bip32Path[5];
 
   bip32Path[0] = 44  | 0x80000000;
   bip32Path[1] = 283 | 0x80000000;
-  bip32Path[2] = 0   | 0x80000000;
+  bip32Path[2] = accountId | 0x80000000;
   bip32Path[3] = 0;
   bip32Path[4] = 0;
   os_perso_derive_node_bip32(CX_CURVE_Ed25519, bip32Path, sizeof(bip32Path) / sizeof(bip32Path[0]), privateKeyData, NULL);
-}
-
-void
-algorand_private_key(cx_ecfp_private_key_t *privateKey)
-{
   cx_ecfp_init_private_key(CX_CURVE_Ed25519, privateKeyData, 32, privateKey);
+
+  os_memset(bip32Path, 0, sizeof(bip32Path));
+  os_memset(privateKeyData, 0, sizeof(privateKeyData));
 }
 
-void
-algorand_public_key(uint8_t *buf)
+int
```

Comment:
Since a length is returned, `size_t` could be used instead of `int`. And even if `int`, it should be a C99-style int for portability i.e. `uiint[8|16|32]_t`

Reply:
Fixed using `size_t` that is the actual meaning of the return value.